src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,81 @@
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { setCurrentTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [loaded, setLoaded] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const { userId, id, title, completed } = currentTodo as Todo;
```

Comment:
The destructuring of `currentTodo` assumes it is always defined. Consider adding a check to ensure `currentTodo` is not null before destructuring to avoid potential runtime errors.